exporter/azuremonitorexporter/log_to_envelope.go

```diff
@@ -0,0 +1,96 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuremonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter"
+
+import (
+	"time"
+
+	"github.com/microsoft/ApplicationInsights-Go/appinsights/contracts"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	traceIdTag      = "TraceId"
+	spanIdTag       = "SpanId"
+	categoryNameTag = "CategoryName"
+)
+
+var severityLevelMap = map[string]contracts.SeverityLevel{
+	"Verbose":     contracts.Verbose,
+	"Information": contracts.Information,
+	"Warning":     contracts.Warning,
+	"Error":       contracts.Error,
+	"Critical":    contracts.Critical}
+
+type logPacker struct {
+	logger *zap.Logger
+}
+
+func (packer *logPacker) LogRecordToEnvelope(logRecord pdata.LogRecord) *contracts.Envelope {
+	envelope := contracts.NewEnvelope()
+	envelope.Tags = make(map[string]string)
+	envelope.Time = toTime(logRecord.Timestamp()).Format(time.RFC3339Nano)
+
+	data := contracts.NewData()
+
+	messaageData := contracts.NewMessageData()
+	messaageData.Properties = make(map[string]string)
+
+	messaageData.SeverityLevel = packer.toAiSeverityLevel(logRecord.SeverityText())
+
+	messaageData.Message = logRecord.Body().StringVal()
+
+	hexTraceId := logRecord.TraceID().HexString()
+	messaageData.Properties[traceIdTag] = hexTraceId
+	envelope.Tags[contracts.OperationId] = hexTraceId
+
+	messaageData.Properties[spanIdTag] = logRecord.SpanID().HexString()
+
+	messaageData.Properties[categoryNameTag] = logRecord.Name()
+	envelope.Name = messaageData.EnvelopeName("")
```

Comment:
Does it matter that this an empty string?

Reply:
It's fine